index.js

```diff
@@ -50,56 +50,74 @@ NYC.prototype._createInstrumenter = function () {
   })
 }
 
-NYC.prototype.cleanup = function () {
-  if (!process.env.NYC_CWD) rimraf.sync(this.tmpDirectory())
+NYC.prototype.addFile = function (filename) {
+  var instrument = true
+  var relFile = path.relative(this.cwd, filename)
+
+  // only instrument a file if it's not on the exclude list.
+  for (var i = 0, exclude; (exclude = this.exclude[i]) !== undefined; i++) {
+    if (exclude.test(relFile)) {
+      instrument = false
+      break
+    }
+  }
+
+  var content = stripBom(fs.readFileSync(filename, 'utf8'))
+
+  if (instrument) {
+    content = this.instrumenter.instrumentSync(content, './' + relFile)
+  }
+
+  return {
+    instrument: instrument,
+    content: content,
+    relFile: relFile
+  }
 }
 
-NYC.prototype._wrapRequire = function () {
+NYC.prototype.addAllFiles = function () {
   var _this = this
 
-  // any JS you require should get coverage added.
-  require.extensions['.js'] = function (module, filename) {
-    var instrument = true
-    var content = fs.readFileSync(filename, 'utf8')
-
-    // only instrument a file if it's not on the exclude list.
-    var relFile = path.relative(_this.cwd, filename)
-    for (var i = 0, exclude; (exclude = _this.exclude[i]) !== undefined; i++) {
-      if (exclude.test(relFile)) {
-        instrument = false
-        break
-      }
-    }
+  this._createOutputDirectory()
 
-    if (instrument) {
-      content = _this.instrumenter.instrumentSync(
-        content,
-        './' + relFile
+  glob.sync('**/*.js').forEach(function (filename) {
+    var obj = _this.addFile(filename)
```

Comment:
We are now reading full file contents of files that should be excluded. Potentially quite a painful performance hit.